src/managers/config.py

```diff
@@ -256,56 +271,112 @@ def scram_properties(self) -> list[str]:
-            f'listener.name.{self.internal_listener.name.lower()}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";'
-        ]
-        client_scram = [
-            auth.name for auth in self.client_listeners if auth.protocol.startswith("SASL_")
+            f'listener.name.{listener_name}.{listener_mechanism}.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";',
+            f"listener.name.{listener_name}.sasl.enabled.mechanisms={self.internal_listener.mechanism}",
         ]
-        for name in client_scram:
-            scram_properties.append(
-                f'listener.name.{name.lower()}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";'
+
+        for auth in self.client_listeners:
+            if not auth.mechanism.startswith("SCRAM"):
+                continue
+            scram_cfg = textwrap.dedent(
+                f"""\
+                listener.name.{auth.name.lower()}.{auth.mechanism.lower()}.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";
+                listener.name.{auth.name.lower()}.sasl.enabled.mechanisms={auth.mechanism}
+                """
             )
+            scram_properties.append(scram_cfg)
 
         return scram_properties
 
     @property
-    def security_protocol(self) -> AuthMechanism:
+    def oauth_properties(self) -> list[str]:
+        """Builds the properties for the oauth listener.
+
+        Returns:
+            list of oauth properties to be set.
+        """
+        oauth_properties: list[str] = []
+        oauth_listener = None
+        for listener in self.client_listeners:
+            if listener.mechanism.startswith("OAUTH"):
+                oauth_listener = listener
+                break
+
+        if oauth_listener:
+            # use jwks validation if jwt token, otherwise use introspection validation
+            validation_cfg = (
+                f'oauth.jwks.endpoint.uri="{self.state.oauth.jwks_endpoint}"'
+                if self.state.oauth.jwt_access_token
+                else f'oauth.introspection.endpoint.uri="{self.state.oauth.introspection_endpoint}"'
+            )
+
+            # TODO: add a truststore with public cacerts for oauth listener
+            oauth_cfg = textwrap.dedent(
+                f"""\
+                listener.name.{oauth_listener.name.lower()}.{oauth_listener.mechanism.lower()}.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \\
+                    oauth.client.id="kafka" \\
+                    oauth.valid.issuer.uri="{self.state.oauth.issuer_url}" \\
+                    {validation_cfg} \\
+                    oauth.username.claim="email" \\
```

Comment:
**todo** This may be one of the most important configuration, and I believe this may need to be configured somehow. I would possibly just factor this out into a constant, attribute or something to provide a bit more exposure to this. Right now it is a bit buried in this configuration list

Reply:
Yeah, I wonder if an 'easy' breaking up of this might be to take out the entire definition of `oauth_cfg` and put that in a function, that takes `oauth_listener` as an arg, so we only call `self.generate_oauth_listener_config(listener=oauth_listener)` or something.

Reply:
I extracted the claim, and fallback claim into two variables, but did not create a separate method. After the suggestion above to simplify this method, I thought that wouldn't be necessary anymore.